swatch-producer-azure/src/main/java/com/redhat/swatch/azure/kafka/streams/BillableUsageAggregate.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.azure.kafka.streams;
+
+import com.redhat.swatch.azure.openapi.model.BillableUsage;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@RegisterForReflection
+@Getter
+@Setter
+@EqualsAndHashCode
+@Slf4j
+public class BillableUsageAggregate {
+
+  private BigDecimal totalValue = new BigDecimal(0);
+  private OffsetDateTime windowTimestamp;
+  private UUID aggregateId = UUID.randomUUID();
```

Comment:
Nitpick: this incurs an unnecessary performance penalty on deserialization. I'd suggest creating a new factory method and using it in `StreamTopologyProducer`, leaving the default constructor alone.